### DISCUSSION TITLE
Working / not working device list

### DISCUSSION THREAD

**Discussion by the-butterfry** [syssi/esphome-jbd-bms#17] [General]
This is really great, thanks for your work on this. A few years ago it was pretty tough to get these guys reporting. I'll be converting a few of them over now.

Starting this thread for a list of the various models that are known to work or not. Please provide detail if not.

**JBD-SP10S009-L8S-100A-B-K** bluetooth: **yes** UART: **untested**
Notes: worked first try.

**Comment by it-koncept** [comment]
Thank you Sebastian for awesome software!
JBD-SP15S020-P13S-60A-B FW 1.3  bluetooth: untested UART: Yes
Notes: All sensors seem to work but the switch does not

**Reply by syssi** [reply]
The switches aren't implemented yet: https://github.com/syssi/esphome-jbd-bms/blob/main/components/jbd_bms/switch/jbd_switch.cpp#L12

**Reply by it-koncept** [reply]
Sorry I didn't realize that - but everything else works perfectly thanks for sharing your work 🥇

**Reply by syssi** [reply]
It's work in progress. The most important controls will be available at some point.

**Comment by mobilex-Godelmann** [comment]
JBD-SP18S-100A-086 Bluetooth: yes UART: untested

Great work! As a total beginner I had to figure out how the configuration worked, but eventually managed to get it done.
A more beginner friendly readme would be most welcome, let me know if I can support here.

Best regards,
Floke

**Reply by syssi** [reply]
I'm happy about every kind of support! Feel free to provide a step by step guide / tutorial.

**Comment by teunito** [comment]
JDB-SP16S020-100A Bluetooth: yes UART: untested
All sensors seem to work. The switches are working with https://github.com/syssi/esphome-jbd-bms/pull/36

The BMS is build in a Basen 25,6 V 100 A battery pack (BG-24100)
Reported as Device model: BSJBD202210-28-005 (seems to be Date of manufacture: 2022.10.28).

**Reply by syssi** [reply]
I've updated the list of supported devices.

**Comment by Microguy31** [comment]
Thanks for this great piece of software!
[JBD-SP22S003](https://fr.aliexpress.com/item/1005004675668215.html?spm=a2g0o.order_list.order_list_main.5.14735e5byVH5R9&gatewayAdapt=glo2fra) bluetooth: **yes** UART: **untested** 
All sensors work. Would like to try the switches asap :-).
Work on 16S 90Ah LiFePo4 battery and ESP32,Mqtt Server and Node-Red on M8S+ Armbian.
edit: works well too with "WT32-ETH01" Ethernet module.

**Reply by syssi** [reply]
@Microguy31 Could you provide the device model reported by the firmware (the state of the `device_model` text sensor)? I would like to add your device to the list of supported devices. The identifier provides details about the specs (supported cells + max. current) of your device.

**Reply by Microguy31** [reply]
The device model reported by the firmware is : SP22S003AL22S40A.

**Reply by syssi** [reply]
Thanks! I've updated the list of supported devices.

**Reply by TheNexter** [reply]
@Microguy31 Hi men, have you try the latest version with [JBD-SP22S003](https://fr.aliexpress.com/item/1005004675668215.html?spm=a2g0o.order_list.order_list_main.5.14735e5byVH5R9&gatewayAdapt=glo2fra)? the switch to enable / disable the charge must be there if i read correctly the patch note ;)

**Comment by johkn** [comment]
JBD-SP22S003A-L22S bluetooth: yes UART: untested
Sensors: work
Switches: work state and command (after changing @main to @add-controls and coment balancer switch) 
Binary sensors: work 

System: ESP32 MQTT Nodered Iobroker 

Thanks for this Software!

**Reply by syssi** [reply]
Do you own the 40A, 50A or 60A model? How many cells are supported?

**Reply by johkn** [reply]
My setup 50A and 16 cell (6-22 cell)

**Comment by cctweaker** [comment]
JBD-AP21S001-L21S-200A-B Bluetooth: yes UART: untested
Sensors: work, reporting correctly
Binary sensors: work, reporting correctly
Switches: untested yet, they are shown disabled, can not test until adding the second battery bank

Thank you very much for your code and time!
Just reporting that it works with the JBD-AP21S001-L21S-200A-B BMS via bluetooth.
Aithinker ESP32-S module is reporting via HA api.
1 BMS at the moment but I used the test-many-ble-connections branch because I will add a second BMS & battery bank soon(-ish).

**Reply by cctweaker** [reply]
Soon-ish became 1 month but today I managed to connect the second BMS and battery bank.
Everything is reported correctly. Has sporadic disconnects when all values are shown as Unknown but reconnects on its own.
test-many-ble-connections branch doesn't seem to have functional switches for charging/discharging.

**Reply by syssi** [reply]
I've pushed some improvements and merged the feature into main.

**Reply by cctweaker** [reply]
Right!
Just switched to latest main branch, works like a champ, charging and discharging switches too.

**Comment by baerrs** [comment]
ONYX RCR e-bike
JBD-SP25S003-P20S-100A-B Bluetooth: yes UART: untested
Sensors: work, reporting correctly
Binary sensors: work, reporting correctly
Switches: untested yet, they are shown
HA API: yes, MQTT: untested

I am getting a repeating error:  
```
[02:44:45][W][component:204]: Component esp32_ble took a long time for an operation (0.24 s).
[02:44:45][W][component:205]: Components should block for at most 20-30ms.
```

I have not looked into this at all.  
```
[02:42:04][D][jbd_bms_ble:263]:   Device model: JBD-SP25S003-P20S-100A-B
[02:42:04][D][sensor:094]: 'jbd-bms-ble total voltage': Sending state 75.58000 V with 2 decimals of accuracy
[02:42:04][D][sensor:094]: 'jbd-bms-ble current': Sending state 0.00000 A with 1 decimals of accuracy
[02:42:04][D][sensor:094]: 'jbd-bms-ble power': Sending state 0.00000 W with 1 decimals of accuracy
[02:42:04][D][sensor:094]: 'jbd-bms-ble charging power': Sending state 0.00000 W with 2 decimals of accuracy
[02:42:04][D][sensor:094]: 'jbd-bms-ble discharging power': Sending state 0.00000 W with 2 decimals of accuracy
[02:42:04][D][sensor:094]: 'jbd-bms-ble capacity remaining': Sending state 29.10000 Ah with 2 decimals of accuracy
[02:42:04][D][sensor:094]: 'jbd-bms-ble nominal capacity': Sending state 41.00000 Ah with 2 decimals of accuracy
[02:42:04][D][sensor:094]: 'jbd-bms-ble charging cycles': Sending state 21.00000  with 0 decimals of accuracy
[02:42:04][D][jbd_bms_ble:290]:   Date of manufacture: 2022.8.16
[02:42:04][D][sensor:094]: 'jbd-bms-ble balancer status bitmask': Sending state 0.00000  with 0 decimals of accuracy
[02:42:04][D][sensor:094]: 'jbd-bms-ble errors bitmask': Sending state 0.00000  with 0 decimals of accuracy
[02:42:04][D][text_sensor:064]: 'jbd-bms-ble errors': Sending state ''
[02:42:04][D][sensor:094]: 'jbd-bms-ble software version': Sending state 2.00000  with 1 decimals of accuracy
[02:42:04][D][sensor:094]: 'jbd-bms-ble state of charge': Sending state 71.00000 % with 0 decimals of accuracy
[02:42:04][D][sensor:094]: 'jbd-bms-ble operation status bitmask': Sending state 3.00000  with 0 decimals of accuracy
[02:42:04][D][sensor:094]: 'jbd-bms-ble battery strings': Sending state 20.00000  with 0 decimals of accuracy
```

This is the example ymal, with the MAC address changed and removed the cells above 20 (I only have 20).  

Thanks for the great work !!

**Reply by syssi** [reply]
Thanks for your feedback! Please ignore the warning.

**Reply by baerrs** [reply]
I'm not sure if this is normal or not, it doesn't seem normal to me.
![image](https://github.com/syssi/esphome-jbd-bms/assets/56986668/4ee9ccb4-6594-47d6-974e-b17c81281335)

**Reply by syssi** [reply]
Please create a new issue and provide a log. It should include multiple `balancer bitmask` measurements.

**Reply by baerrs** [reply]
I can do that.. what logs are you looking for?  

I wanted to share an image of the HA interface:
![image](https://github.com/syssi/esphome-jbd-bms/assets/56986668/3be8cf65-725d-4be8-8621-0ab65a056cba)

**Comment by chipsi** [comment]
JBD-SP22S003B-L21S-100A Bluetooth: yes UART: yes
Sensors: work, reporting correctly
Binary sensors: work, reporting correctly

JBD-SP22S003B-L22S-100A Bluetooth: yes UART: yes
Sensors: work, reporting correctly
Binary sensors: work, reporting correctly

JBD-AP20S003-L20S-200A-B Bluetooth: yes UART: yes
Sensors: work, reporting correctly
Binary sensors: work, reporting correctly

BLE mode sometimes communication gets stuck. UART mode is OK. 

Great work

**Comment by RudolfRendier** [comment]
Bought battery based on recommendation by @teunito. Still works.
Basen Green 24V 100Ah.
SP22S003BL8S100A Software version 7.6
Works fine via BLE, both sensors, binary sensors.

**Reply by syssi** [reply]
Is your battery pack also called BG-24100?

**Comment by stalniy** [comment]
Batery 24V, 100AH, JBD-SP16S020-L8S

Works! Thank you for this extension!
<img width="928" alt="Screenshot 2025-02-02 at 06 37 45" src="https://github.com/user-attachments/assets/3c305e13-7652-48ed-a581-aa995949a384" />

**Comment by 568541318** [comment]
SP17S005P17S40A  Bluetooth: untested UART: yes
![image](https://github.com/user-attachments/assets/6b989929-5285-441b-9163-ddc617aff59a)

**Reply by SurampudiAruna** [reply]
@568541318 Were the hardware connections between the JBD 17S BMS and the ESP32 the same as described in the README? I followed the wiring in the documentation, but I’m not getting any data response.

**Reply by 568541318** [reply]
Please check the UART hardware connections. Note that the JBD RX pin is connected to the ESP32 TX pin, and the JBD TX pin is connected to the ESP32 RX pin.

**Comment by WP-Rue** [comment]
Thank you very much!
Working with WT-32 LAN-Connection and 3x JBD17S005 40A via bluetooth!
Greetings Rue

**Reply by syssi** [reply]
Could you double check the model name of your BMS? Are you sure your model is called JBD17S05?

**Reply by WP-Rue** [reply]
![image](https://github.com/user-attachments/assets/2c904d78-fd6b-4e0e-8fa7-aa6615f7d6db)
One zero missing 😉

**Reply by syssi** [reply]
If you use this project there should be a "retrieve hardware version" button which prints a model name to the log / populates the text sensor "device model". Could you provide this value?

**Reply by WP-Rue** [reply]
Haken I Push this Button? 
I use this software without Home Assistant; only MQTT and than Node Red..

**Reply by syssi** [reply]
If this button is part of your YAML configuration you could send `PRESS` to the topic.

**Reply by WP-Rue** [reply]
I will try later this afternoon ..

**Reply by syssi** [reply]
Take your time. It's not urgent. It's just about completeness. :-)

**Reply by WP-Rue** [reply]
Ok- I sent "PRESS" to the topic "jbd-bms-ble/button/retrieve_hardware_version" - 

In debug messages I can find:
...bms_ble:262]:   Device model: [0m
So- no entry.
Manufakture date is 27.4.2024.
Greetings 
Rue

**Reply by syssi** [reply]
Okay. The manufacturer did not populate the register. So the sticker is all we have.

**Reply by WP-Rue** [reply]
Expand now to the fourth BMS with BLE- still working with W32-ETH01.
Thanks a lot!

**Comment by takbo3000** [comment]
Liontron LiFePO4 Lithium 12.8V 150Ah works with Bluetooth/HomeAssistant. Thank you very much!

**Comment by redwoodtree** [comment]
Epoch Essentials LiFePO4 12.8V 460Ah working with Bluetooth & Home Assistant, out of the box, with no issues. Thanks for this great integration, amazing!!

**Reply by syssi** [reply]
I've added your battery pack to the README: https://github.com/syssi/esphome-jbd-bms/commit/f79d2a218d2025660f3f31a4434050127cea4248

**Comment by Gerhard34** [comment]
JBD-SP04S034-100A Bluetooth: yes UART: untested
Produktionsdatum: 2022-2-17
BMS-Version: 8.0

Allerdings OHNE password (das sollte gem.  [Beitrag](https://github.com/syssi/esphome-jbd-bms/issues/145#issuecomment-3137697400) erforderlich sein?!?)

**Comment by johnkims-254** [comment]
JBS BMS Model: JBD-UP16S019-200A 
Bluetooth: yes UART: untested
Production Date: 2025-12-06
BMS-Firmware Version: JBD.14.00.03
BMS Module Version: JBD.BW03.01.011
Battery: AMIBA HES05WT Model (100Ah LiFePO4)

Many thanks @syssi for this great HA integration. I have a slightly outdated inverter without BMS port and this solution helped to make my HA energy dashboard more complete after switching from lead-acid to Lithium battery - previously without BMS communication, data from the battery was unreliable.

Some useful features that I see in the JBD BMS mobile App that are not being pulled into my HA are:
 - State of Health
 - Protection status
 - Status of various MOSFETs
 
 Assuming it is not the limitation of my understanding, inclusion of the above data would be great. Otherwise, loving the integration!!

**Reply by syssi** [reply]
Thanks for your feedback! Feel free to create an issue per missing feature!

**Reply by johnkims-254** [reply]
Done!

**Comment by bigskookum** [comment]
Model name: SP20S002-P16S-100A
Bluetooth: Yes, UART: Untested
Date of manufacture: 2018.8.23
software version 2.0

Other versions unknown, unless there is a way to extract them with this project.

Using Nissan Leaf cells to power an ATV and running strong for almost a decade.

**Comment by Marco82tn** [comment]
Good morning, has it also been tested with the ESP32-ETH01 version? I have jbd bms UP16S008

**Reply by syssi** [reply]
Yes. The ESP-ETH01 is a good choice!

**Reply by Marco82tn** [reply]
I am an electronics technician and a beginner with ESP; I am trying to understand how the programming works. I’ve tried out some internal components in ESPHome and they work, but I don’t understand how to load and adapt external components. I saw the string that needs to be added—it seems to load, but then nothing happens. I’m sure the configuration needs to be modified for my specific ESP32 model.

**Reply by syssi** [reply]
Please create a new issue and post your YAML configuration. I will provide some guidance!

**Reply by Marco82tn** [reply]
I believe the UP16s008 version should use Stefan's modified version.

**Comment by Marco82tn** [comment]
jbd bms UP16S008 Firmware 2.4 UART:YES

WT32 ETH01, RS485 (chip max13487E) power supply +5V
code @[stefan_04] (https://github.com/stefan064)

**Comment by freeesty** [comment]
DB24SA03P21S80A v7.c
Manufact. date 28.07.2025

working too.
